### PR TITLE
chore(lint): extend linting to all modules and tools

### PR DESCRIPTION
**Pull request by alfaproject**
**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?**

```
[ ] Bugfix
[ ] Feature
[x] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[x] Build related changes
[x] CI related changes
[ ] Other... Please describe:
```

**What is the current behavior?**
Not all modules and tools are being linted.
Relates to #12464

**What is the new behavior?**
All modules and tools are linted.

**Does this PR introduce a breaking change?**

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -148,14 +148,15 @@ gulp.task('lint', ['check-tests', 'format:enforce', 'tools:build'], () => {
...
+        './modules/**/*.ts',
+        './tools/**/*.ts',
         './*.ts',
+        '!**/*.d.ts',
+        '!**/*.ngfactory.ts',
       ])
       .pipe(tslint({
         tslint: require('tslint').default,
         configuration: tslintConfig,
-        rulesDirectory: 'dist/tools/tslint',
```
Comment (alfaproject) [inline gulpfile.js]: I moved this to `tslint.json` to enable other tools to know about the custom rules. (Like Atom or VSCode plugins.)

File: gulpfile.js [hunk 2]
```diff
@@ -148,14 +148,15 @@ gulp.task('lint', ['check-tests', 'format:enforce', 'tools:build'], () => {
       .src([
         // todo(vicb): add .js files when supported
         // see https://github.com/palantir/tslint/pull/1515
-        'modules/@angular/**/*.ts',
-        'modules/benchpress/**/*.ts',
+        './modules/**/*.ts',
+        './tools/**/*.ts',
         './*.ts',
+        '!**/*.d.ts',
+        '!**/*.ngfactory.ts',
```
Comment (vicb) [inline gulpfile.js]: It might be valuable (to check that we correctly disable the checks)

File: gulpfile.js (hunk 1 above)
Comment (alfaproject) [inline gulpfile.js]: Then nothing changes, because if you use the gulp task directly, it will still build the tools before it runs tslint. (;

Comment (alfaproject) [inline gulpfile.js]: Yes, I suppose so. Will enable then.

Comment (alfaproject) [inline gulpfile.js]: The thing is the other tools are already failing, so nothing really changes. But if it's compiled, then at least they can be used.

File: gulpfile.js (hunk 2 above)
Comment (alfaproject) [inline gulpfile.js]: I've enabled them to see what happens, but then TSLint will bitch about the generated files not having the copyright header. ):

```
[19:07:10] Starting 'lint'...
benchmarks/src/tree/ng2_ftl/app.ngfactory.ts[1, 1]: missing copyright header

benchmarks/src/tree/ng2_ftl/ng_if.ngfactory.ts[8, 3]: module-private member _expr_0 must be annotated @internal
benchmarks/src/tree/ng2_ftl/ng_if.ngfactory.ts[1, 1]: missing copyright header

benchmarks/src/tree/ng2_ftl/tree.ngfactory.ts[1, 1]: missing copyright header

benchmarks/src/tree/ng2_ftl/tree_host.ngfactory.ts[1, 1]: missing copyright header

benchmarks/src/tree/ng2_static_ftl/app.ngfactory.ts[1, 1]: missing copyright header

benchmarks/src/tree/ng2_static_ftl/tree_branch.ngfactory.ts[1, 1]: missing copyright header

benchmarks/src/tree/ng2_static_ftl/tree_leaf.ngfactory.ts[1, 1]: missing copyright header

benchmarks/src/tree/ng2_static_ftl/tree_root.ngfactory.ts[1, 1]: missing copyright header
```

Comment (vicb) [inline gulpfile.js]: Ok, please add a comment explaining why here then. Thanks.

File: gulpfile.js [hunk 3]
```diff
@@ -142,20 +142,23 @@ gulp.task('check-tests', function() {
...
       .src([
         // todo(vicb): add .js files when supported
         // see https://github.com/palantir/tslint/pull/1515
-        'modules/@angular/**/*.ts',
-        'modules/benchpress/**/*.ts',
+        './modules/**/*.ts',
+        './tools/**/*.ts',
         './*.ts',
+
+        // Ignore auto generated files due to lack of copyright header
```
Comment (vicb) [inline gulpfile.js]: `// Ignore generated files due to lack of copyright header`

would be nice if those were "auto" generated :)

File: tslint.json [hunk 4]
```diff
@@ -1,10 +1,12 @@
...
-      "no-duplicate-variable": true,
-      "semicolon": [true],
-      "variable-name": [true, "ban-keywords"]
-    }
+  "rulesDirectory": ["./dist/tools/tslint"],
+  "rules": {
+    "require-internal-with-underscore": true,
+    "duplicate-module-import": true,
+    "enforce-copyright-header": true,
+
```
Comment (vicb) [inline tslint.json]: remove blank line

File: tools/@angular/tsc-wrapped/test/typescript.mocks.ts [hunk 5]
```diff
@@ -73,12 +73,12 @@ export class MockNode implements ts.Node {
 
 export class MockIdentifier extends MockNode implements ts.Identifier {
```
Comment (vicb) [inline tools/@angular/tsc-wrapped/test/typescript.mocks.ts]: please revert this file - not angular API

**Comment by vicb** [comment]
A few nits. Thanks very much for the changes.
Should be ready to merge once the minor comments are addressed.

File: gulpfile.js (hunk 3 above)
Comment (alfaproject) [inline gulpfile.js]: Indeed, but one of the `ngfactory` files actually has another error. I'll add a todo note, and maybe we can tackle it separately.

**Comment by alfaproject** [comment]
@vicb everything should be addressed now. (;

File: gulpfile.js [hunk 6]
```diff
@@ -143,20 +143,27 @@ gulp.task('check-tests', function() {
...
-        'modules/@angular/**/*.ts',
-        'modules/benchpress/**/*.ts',
+        './modules/**/*.ts',
+        './tools/**/*.ts',
         './*.ts',
+
+        // Ignore TypeScript mocks because it's not managed by us
+        '!./tools/@angular/tsc-wrapped/test/typescript.mocks.ts',
+
+        // Ignore auto generated files due to lack of copyright header
```
Comment (vicb) [inline gulpfile.js]: remove "auto"

File: gulpfile.js [hunk 7]
```diff
@@ -143,20 +143,27 @@ gulp.task('check-tests', function() {
...
       .src([
         // todo(vicb): add .js files when supported
         // see https://github.com/palantir/tslint/pull/1515
-        'modules/@angular/**/*.ts',
-        'modules/benchpress/**/*.ts',
+        './modules/**/*.ts',
+        './tools/**/*.ts',
         './*.ts',
+
+        // Ignore TypeScript mocks because it's not managed by us
```
Comment (vicb) [inline gulpfile.js]: ?

**Comment by vicb** [comment]
Please rebase and fix the few comments. Thanks.

File: gulpfile.js (hunk 7 above)
Comment (alfaproject) [inline gulpfile.js]: You said to revert that file because it's not our API. I assumed we don't want to lint it unless you mean any particular rule?

**Comment by alfaproject** [comment]
I've just rebased this. Should be mergeable again. (:

**Comment by vsavkin** [comment]
merged https://github.com/angular/angular/commit/5494169fb43932c67822a8f229afcfb5d67a95b1

**Pull request closed without merging**